File: Graph/1260_DFS_N_BFS.py
import sys

input = sys.stdin.readline


N, M, V = map(int, input().strip().split())
# 무향 그래프 만들기
graph = [[] for _ in range(N+1)]
for _ in range(M):
    spot, node = map(int, input().strip().split())
    graph[spot].append(node)
    graph[node].append(spot)

stack = [V]
queue = [V]
dfs_visited = [0] * (N+1)
bfs_visited = [0] * (N+1)
dfs_visited[V] = 1
res = []            # 결과 출력용
# DFS
while stack:
    dfs = stack.pop()
    dfs_visited[dfs] = 1
    if dfs not in res:
        res.append(dfs)     # 경로에 가본적 없으면 저장
    graph[dfs].sort(reverse=1)
    for i in graph[dfs]:
        if not dfs_visited[i]:
            stack.append(i)
print(*res)
# BFS
bfs_visited[V] = 1
while queue:
    bfs = queue.pop(0)
    print(bfs, end=' ')
    graph[bfs].sort()
    for i in graph[bfs]:
        if not bfs_visited[i]:
            bfs_visited[i] = 1
            queue.append(i)
print()

File: BruteForce/1260_dfs_n_bfs.js
const fs = require("fs");

const settings = () => {
    // const input = fs.readFileSync("input.txt").toString().trim().split("\n");
    const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");
    const [N, M, V] = input[0].split(" ").map(Number);
    const graph = [];
    for (let i =0;i<=N;i++){
        graph.push([]);
    }
    input.slice(1,).forEach(e => {
        const [start,end] = e.split(" ").map(Number);
        graph[start].push(end);
        graph[end].push(start);
    })
    // console.log(graph)
    return [N, V,graph];
}

const main = (N, start, graph) => {
    const dfs = (start) => {
        const visited = Array(N+1).fill(0);
        const answer = [];
        const stack = [start];
        // visited[start] = 1;
        while (stack.length > 0) {
            // console.log(stack);
            const spot = stack.pop();
            if (visited[spot]) continue;
            visited[spot] = 1
            answer.push(spot);
            graph[spot].sort((a,b) => b-a);
            for (let node of graph[spot]) {
                // if (!visited[node]){
                    // visited[node] = 1;
                stack.push(node);
                // }
            }
        }
        console.log(...answer);
    }

    const bfs = (start) => {
        const visited = Array(N+1).fill(0);
        const answer = [];
        const queue = {};
        let [front, rear] = [0, 0];
        queue[rear++] = start;
        visited[start] = 1;
        while (rear - front > 0) {
            const spot = queue[front++];
            answer.push(spot);
            graph[spot].sort((a,b) => a-b);
            for (let node of graph[spot]) {
                if (!visited[node]) {
                    visited[node] = 1;
                    queue[rear++] = node;
                }
            }
        }
        console.log(...answer);
    }

    dfs(start);
    bfs(start);
}

const [N ,start, graph] = settings();
main(N, start, graph);